File: years/2015/01/part1.ts
import input from "./input.txt";

const lines = input.split("\n");

let floor = 0;
for (const c of [...lines[0]]) {
    if (c === "(") {
        floor++;
    } else if (c === ")") {
        floor--;
    }
}
console.log(floor);

File: years/2025/01/part1.hs
import Control.Arrow

solve = words >>> map parseCommand >>> scanl go 50 >>> filter (== 0) >>> length
  where
    go a b = (a + b) `mod` 100

parseCommand :: String -> Integer
parseCommand (x : xs)
    | x == 'L' = -num
    | x == 'R' = num
  where
    num = read xs

main = readFile "input.txt" >>= (print . solve)
